### DISCUSSION TITLE
Trouble displaying security camera feed in OpenBSD

### DISCUSSION THREAD

**Discussion by kd4e** [qutebrowser/qutebrowser#8196] [Q&A]
I can see the feed using qutebrowser with Linux.

On this computer I'm using OpenBSD (`uname - a` shows "7.4 GENERIC.MP#2 and64")

Last night I tried and got a blank screen.

I just ran pkg_add -uvi and tried again and got the following ...

![openbsd-qutebrowser-camera1](https://github.com/qutebrowser/qutebrowser/assets/74081053/7f33f006-04e7-4781-8d83-b9fbe4622726)

Ideas, please?

Thanks

**Comment by The-Compiler** [comment]
Please try with `--temp-basedir` and show your full `:version` info.

**Reply by kd4e** [reply]
`--temp-basedir` gives the same result. `pkg_info qutebrowser` says 3.0.0

**Reply by The-Compiler** [reply]
I'm asking for the full `:version` info because I need more information than just the qutebrowser version...

**Reply by kd4e** [reply]
I'm afraid I don't know how to request that information.

How do I request of OpenBSD additional version detail?

All I know is `uname -a` or `uname -r`.

I tried `more /var/run/dmesg.boot` and it dumped a bunch of output - but at the top just 7.4 for OpenBSD.

**Reply by The-Compiler** [reply]
You run `:version` in qutebrowser by typing just that, like any other qutebrowser command.

**Comment by kd4e** [comment]
qutebrowser v3.0.0
Git commit: 
Backend: QtWebEngine 5.15.10, based on Chromium 87.0.4280.144
Qt: 5.15.10

CPython: 3.10.13
PyQt: 5.15.9

Qt wrapper: PyQt5 (via override)

colorama: no
jinja2: 3.1.2
pygments: no
yaml: 5.4.1
adblock: 0.6.0
objc: no
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.6
PyQt5.QtWebKitWidgets: yes
PyQt5.sip: 6.7.10
pdf.js: no
sqlite: 3.42.0
QtNetwork SSL: LibreSSL 3.8.2

Style: QFusionStyle
Platform plugin: xcb
OpenGL: Intel, 4.6 (Compatibility Profile) Mesa 22.3.7
Platform: OpenBSD-7.4-amd64-64bit-ELF, 64bit
Frozen: False
Imported from /usr/local/lib/python3.10/site-packages/qutebrowser
Using Python from /usr/local/bin/python3.10
Qt library executable path: /usr/local/lib/qt5/libexec, data path: /usr/local/lib/qt5
OS Version: OpenBSD dmc1.my.domain 7.4 GENERIC.MP#2 amd64 amd64

Paths:
cache: /root/.cache/qutebrowser
config: /root/.config/qutebrowser
data: /root/.local/share/qutebrowser
runtime: /tmp/runtime-root/qutebrowser

Autoconfig loaded: yes
Config.py: no config.py was loaded
Uptime: 0:01:28

**Reply by The-Compiler** [reply]
Looks like OpenBSD [doesn't have PyQt6 yet](https://www.mail-archive.com/ports@openbsd.org/msg121873.html) (3.5 years after its [first release](https://pypi.org/project/PyQt6/#history)...), so you're on Qt 5 with Chromium 87 from 2020.

Chances are that the website you're displaying uses something that's not supported with such an old browser engine. Looking at the `:devtools` console might say more about what exactly, but chances are there isn't much that can be done without OpenBSD updating their PyQt and qutebrowser.

**Reply by kd4e** [reply]
Always something. I wonder why they're stalling on updating PyQt. I'll look for a place to nag them about it. Thanks.

**Reply by The-Compiler** [reply]
Looks like there [was some recent activity](https://www.mail-archive.com/ports@openbsd.org/msg125391.html), FWIW.

**Reply by kd4e** [reply]
Anything in these updates in 7.5 that's encouraging? Python 2.7.18, 3.9.18, 3.10.13, 3.11.8
Qt 5.15.12 (+ kde patches), 6.6.1

**Reply by The-Compiler** [reply]
Newer Python and Qt are always good (modulo the usual bugs in .0 Qt versions) - but without PyQt6, Qt 6 isn't really relevant for qutebrowser.

**Reply by kd4e** [reply]
Just heard back from JustinB "I did not touch it for a few months now. I had several ISP issues.
It is all sorted now and I will resume my work on it soon(ish)."
